File: NextApp/iOS/OrgPortalNext/Core/Model/VideoRepeatSetting.swift
import Foundation

public enum VideoRepeatMode: String, Codable, CaseIterable, Sendable {
    case full
}

public struct VideoRepeatSetting: Codable, Equatable, Sendable {
    public var userId: String
    public var videoId: String
    public var isEnabled: Bool
    public var mode: VideoRepeatMode
    public var repeatStartSeconds: Double?
    public var repeatEndSeconds: Double?

    public init(
        userId: String,
        videoId: String,
        isEnabled: Bool,
        mode: VideoRepeatMode = .full,
        repeatStartSeconds: Double? = nil,
        repeatEndSeconds: Double? = nil
    ) {
        self.userId = userId
        self.videoId = videoId
        self.isEnabled = isEnabled
        self.mode = mode
        self.repeatStartSeconds = repeatStartSeconds
        self.repeatEndSeconds = repeatEndSeconds
    }
}

File: NextApp/Android/core/model/src/main/kotlin/jp/komehyappyo/member/next/core/model/VideoRepeatSetting.kt
package jp.komehyappyo.member.next.core.model

enum class VideoRepeatMode(val rawValue: String) {
    Full("full"),
    ;

    companion object {
        fun fromValue(value: String?): VideoRepeatMode = when (value) {
            Full.rawValue -> Full
            else -> Full
        }
    }
}

data class VideoRepeatSetting(
    val userId: String,
    val videoId: String,
    val isEnabled: Boolean,
    val mode: VideoRepeatMode = VideoRepeatMode.Full,
    val repeatStartSeconds: Double? = null,
    val repeatEndSeconds: Double? = null,
)
